src/magic.c

```diff
@@ -1127,8 +1126,13 @@ double magic_resistance(unit * target)
     curse *c;
     const curse_type * ct_goodresist = 0, *ct_badresist = 0;
     const resource_type *rtype;
-    double probability = u_race(target)->magres;
+    const race *rc = u_race(target);
+    double probability = rc->magres;
+    const plane *pl = rplane(target->region);
 
+    if (rc == get_race(RC_HIRNTOETER) && !pl) {
```

Comment:
Das wäre sicher zukunftssicherer, aber es war viel einfacher so. So lange es Test coverage hat, kann man das aber in Zukunft immer noch ändern, wenn man mal andere Rassen einführen will, die ähnliches Verhalten haben.